### PR TITLE
APEXMALHAR-2176 expressionFunctions for FilterOperator

### PR DESCRIPTION
**Pull request by yogidevendra**
1. added setExpressionFunctionsItem method

### CODE REVIEW

**Comment by yogidevendra** [comment]
@pradeepdalvi @amberarrow Could you please review this?

File: library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java [hunk 1]
```diff
@@ -19,47 +19,48 @@
...
 import com.datatorrent.api.annotation.InputPortFieldAnnotation;
-
 import com.datatorrent.common.util.BaseOperator;
 import com.datatorrent.lib.expression.Expression;
 import com.datatorrent.lib.util.PojoUtils;
 
 /**
- * <b>FilterOperator</b>
- * Filter Operator filter out tuples based on defined condition
+ * <b>FilterOperator</b> Filter Operator filter out tuples based on defined
```
Comment (chinmaykolhatkar) [inline library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java]: Please remove the changes which are not related to this Jira.

File: library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java [hunk 2]
```diff
@@ -127,22 +126,23 @@ public void beginWindow(long windowId)
   protected void createExpression()
   {
     logger.info("Creating an expression for condition {}", condition);
+    expressionFunctions.addAll(optionalExpressionFunctions);
```
Comment (chinmaykolhatkar) [inline library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java]: You need to add only non-null elements to expressionFunctions.

File: library/src/test/java/com/datatorrent/lib/filter/FilterTest.java [hunk 3]
```diff
@@ -173,9 +175,18 @@ public void testFilterError()
...
   }
+  
+  @Test
+  public void testOptionalExpressionFunctions()
+  {
+    filter.setOptionalExpressionFunctions(Arrays.asList(new String[] {"org.apache.commons.lang3.BooleanUtils.*"}));
+    prepareFilterOperator(DummyPublicPOJO.class, "({$}.val == 1)");
+    Assert.assertEquals(6, filter.getExpressionFunctions().size());
+  }
+
```
Comment (chinmaykolhatkar) [inline library/src/test/java/com/datatorrent/lib/filter/FilterTest.java:186]: Can you also add test for setOptionalExpressionFunctionsItem?
Consider the out of bound condition where during createExpression, there will be some null elements in optionalExpressionFunctions.

File: library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java [hunk 4]
```diff
@@ -19,47 +19,48 @@
...
  * 
  */
 @InterfaceStability.Evolving
 public class FilterOperator extends BaseOperator implements Operator.ActivationListener
 {
   private String condition;
-  private List<String> expressionFunctions = new LinkedList<>();
+  
+  private List<String> expressionFunctions;
+  private List<String> optionalExpressionFunctions = new ArrayList<>();
```
Comment (chinmaykolhatkar) [inline library/src/main/java/com/datatorrent/lib/filter/FilterOperator.java]: Would it make sense to rename this to additionalExpressionFunctions?

**Comment by chinmaykolhatkar** [comment]
@yogi, Initial review done. Did you run try this on a real application in YARN and not in local cluster?

**Comment by yogidevendra** [comment]
Yes. We have an example app for this operator which is tried in lab cluster.

**Comment by yogidevendra** [comment]
@chinmaykolhatkar Incorporated review comments. Could you please check and merge this?

**Comment by chinmaykolhatkar** [comment]
@yogidevendra Travis build command is failing for me locally.Tried on fresh git clone and applying ur changes as well. 
Can you please check at ur end whether you get the same issue?

**Comment by yogidevendra** [comment]
@chinmaykolhatkar Travis build command gives BUILD SUCCESSFUL for my local setup. Tried on fresh git clone and pulling this PR on local.
